lib/src/ui/pages/authentication/login/login_notifier.dart

```diff
@@ -50,6 +50,7 @@ class LoginNotifier extends ChangeNotifier {
     _state = _state.copyWithFieldsErrors(
       emailError: _state.email.isEmpty ? emptyFieldError : null,
       passwordError: _state.password.isEmpty ? emptyFieldError : null,
+      isLoading: false,
     );
 
     _state.canSubmit ? onValid() : onInvalid();
```

Comment:
teraz zauważyłem, że walidacja jest niespójna bo canSubmit patrzy czy są różne od null, a walidator czy są puste, dlatego fajnie, żeby ta walidacja była w jednym miejscu a nie dwóch, bo powoduje to takie małe niespójności :)

Reply:
trochę inaczej to działa, walidacja wewnątrz notifiera sprawdza czy **wpisywane przez usera pola** są błędne. W tym konkretnym przypadku jest to sprawdzanie jedynie czy pola są puste, ale ta walidacja może być też rozbudowana o regex etc. Każde pole może mieć inną walidację + kilka pól jednocześnie może być błędne. 

Stan natomiast sprawdza czy jakiekolwiek **pole błędu** jest nullem, żeby ocenić czy walidacja wszystkich pól wpisywanych przez usera zakończyło się błędem

Reply:
ach, masz rację, nie spojrzałem dokładnie :D